.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseMergedExtractor.cs

```diff
@@ -110,8 +110,17 @@ private void AddTo(List<ExtractResult> dst, List<ExtractResult> src, string text
                     //}
                     if (dst[i].IsOverlap(result))
                     {
+                        int nextCursor = i + 1;
+                        var isCrossOverlap = false;
+                        if (nextCursor < dst.Count)
+                        {
+                            if (dst[nextCursor].IsCrossOverlap(result))

```

Comment:
i don't think this should be treated as overlap. "2 hours next" is an incorrect extraction.

Reply:
Also, in one of the modes, the recognition is already correct, but with no resolution.
Wouldn't it be better to just calculate the resolution and fix why the behaviour is different across modes?